frame/collective/src/lib.rs

```diff
@@ -658,6 +653,25 @@ fn get_result_weight(result: DispatchResultWithPostInfo) -> Option<Weight> {
 }
 
 impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	/// The active proposals.
+	pub fn proposals() -> BoundedVec<Bounded<<T as Config<I>>::Proposal>, T::MaxProposals> {
```

Comment:
Is this only used for testing?

```suggestion
	#[cfg(feature = "std")]
	pub fn proposals() -> BoundedVec<Bounded<<T as Config<I>>::Proposal>, T::MaxProposals> {
```

Then it should live behind a std gate, to not accidentally call it.

Reply:
I didn't do it because it might be used outside of testing in external pallets. Should I?

Reply:
External pallets can use it in tests if you `std` gate it. All functions that are not runtime-safe should be gated with that (or `runtime-benchmarks` whichever fits better).